File: namsor/php/src/JapaneseNameKanjiCandidatesExample.php
<?php

namespace OSEG\NamsorExamples;

require_once __DIR__ . '/../vendor/autoload.php';

use SplFileObject;
use Namsor;

$config = Namsor\Client\Configuration::getDefaultConfiguration();
$config->setApiKey("X-API-KEY", "YOUR_API_KEY");

try {
    $response = (new Namsor\Client\Api\JapaneseApi(config: $config))->japaneseNameKanjiCandidates(
        japanese_surname_latin: "Sanae",
        japanese_given_name_latin: "Yamamoto",
        known_gender: "m",
    );

    print_r($response);
} catch (Namsor\Client\ApiException $e) {
    echo "Exception when calling JapaneseApi#japaneseNameKanjiCandidates: {$e->getMessage()}";
}

File: namsor/ruby/src/JapaneseNameKanjiCandidatesExample.rb
require "json"
require "namsor_client"

NamsorClient.configure do |config|
    config.api_key["X-API-KEY"] = "YOUR_API_KEY"
end

begin
    response = NamsorClient::JapaneseApi.new.japanese_name_kanji_candidates(
        "Sanae", # japanese_surname_latin
        "Yamamoto", # japanese_given_name_latin
        "m", # known_gender
    )

    p response
rescue NamsorClient::ApiError => e
    puts "Exception when calling JapaneseApi#japanese_name_kanji_candidates: #{e}"
end
